gradio/components/image_editor.py

```diff
@@ -0,0 +1,292 @@
+    background: Optional[FileData] = None
+    layers: List[FileData] = []
+    composite: Optional[FileData] = None
+
+
+@dataclasses.dataclass
+class Eraser:
+    default_size: int | Literal["auto"] = "auto"
+
+
+@dataclasses.dataclass
+class Brush(Eraser):
+    colors: Union[
+        list[str],
+        str,
+        None,
+    ] = None
+    default_color: Union[str, Literal["auto"]] = "auto"
+    color_mode: Literal["fixed", "defaults"] = "defaults"
+
+    def __post_init__(self):
+        if self.colors is None:
+            self.colors = [
+                "rgb(204, 50, 50)",
+                "rgb(173, 204, 50)",
+                "rgb(50, 204, 112)",
+                "rgb(50, 112, 204)",
+                "rgb(173, 50, 204)",
+            ]
+        if self.default_color is None:
+            self.default_color = (
+                self.colors[0] if isinstance(self.colors, list) else self.colors
+            )
+
+
+@document()
+class ImageEditor(Component):
+    """
+    Creates an image component that can be used to upload images (as an input) or display images (as an output).
+    Preprocessing: passes the uploaded image as a {numpy.array}, {PIL.Image} or {str} filepath depending on `type`.
+    Postprocessing: expects a {numpy.array}, {PIL.Image} or {str} or {pathlib.Path} filepath to an image and displays the image.
+    Examples-format: a {str} local filepath or URL to an image.
+    Demos: image_mod, image_mod_default_image
+    Guides: image-classification-in-pytorch, image-classification-in-tensorflow, image-classification-with-vision-transformers, building-a-pictionary_app, create-your-own-friends-with-a-gan
+    """
+
+    EVENTS = [
+        Events.clear,
+        Events.change,
+        Events.select,
+        Events.upload,
+    ]
+    data_model = EditorData
+
+    def __init__(
+        self,
+        value: str | _Image.Image | np.ndarray | None = None,
+        *,
+        height: int | None = None,
+        width: int | None = None,
+        image_mode: Literal[
+            "1", "L", "P", "RGB", "RGBA", "CMYK", "YCbCr", "LAB", "HSV", "I", "F"
+        ] = "RGBA",
+        sources: Iterable[Literal["upload", "webcam", "clipboard"]] = (
+            "upload",
+            "webcam",
+            "clipboard",
+        ),
+        type: Literal["numpy", "pil", "filepath"] = "numpy",
+        label: str | None = None,
+        every: float | None = None,
+        show_label: bool | None = None,
+        show_download_button: bool = True,
+        container: bool = True,
+        scale: int | None = None,
+        min_width: int = 160,
+        interactive: bool | None = None,
+        visible: bool = True,
+        elem_id: str | None = None,
+        elem_classes: list[str] | str | None = None,
+        render: bool = True,
+        mirror_webcam: bool = True,
+        show_share_button: bool | None = None,
+        _selectable: bool = False,
+        data_mode: Literal["image", "pathline"] = "image",
```

Comment:
I don't think this is supported at the moment, right? Should we remove this param